network/wsNetwork.go

```diff
@@ -1162,16 +1161,24 @@ func (wn *WebsocketNetwork) meshThread() {
 		}
 
 		// TODO: only do DNS fetch every N seconds? Honor DNS TTL? Trust DNS library we're using to handle caching and TTL?
-		dnsAddrs := wn.getDNSAddrs()
-		if len(dnsAddrs) > 0 {
-			wn.log.Debugf("got %d dns addrs, %#v", len(dnsAddrs), dnsAddrs[:imin(5, len(dnsAddrs))])
-			wn.dnsPhonebook.ReplacePeerList(dnsAddrs)
-			mp, ok := wn.phonebook.(*MultiPhonebook)
-			if ok {
-				mp.AddPhonebook(&wn.dnsPhonebook)
+		dnsBootstrapArray := wn.config.DNSBootstrapArray(wn.NetworkID)
+		multiPhonebook := &MultiPhonebook{}
+		dnsStatus := false
+		for _, dnsBootstrap := range dnsBootstrapArray {
+			dnsAddrs := wn.getDNSAddrs(dnsBootstrap)
+			if len(dnsAddrs) > 0 {
+				wn.log.Debugf("got %d dns addrs, %#v", len(dnsAddrs), dnsAddrs[:imin(5, len(dnsAddrs))])
+				dnsPhonebook := &ThreadsafePhonebook{
+					addrs: dnsAddrs,
+				}
+				multiPhonebook.AddPhonebook(dnsPhonebook)
+				dnsStatus = true
+			} else {
+				wn.log.Debugf("got no DNS addrs for network %#v", wn.NetworkID)
 			}
-		} else {
-			wn.log.Debugf("got no DNS addrs for network %#v", wn.NetworkID)
+		}
+		if dnsStatus {
+			wn.phonebook = multiPhonebook
```

Comment:
wn.phonebook needs to be set only once.  There are several sources that contribute to this multi-phonebook and DNSBoostrap results are only one.
I would rewrite to store a map of boostrapIDs to Phonebook.  Then a refresh would be: add/update dnsPhonebooks, mp.AddPhonebook(addedOrUpdated).  We can assume for now we'll never have to REMOVE since thie DNSBootstrapArray is based on a fixed string (from config.json).  The key is you can't just replace wn.phonebook as that wipes out other entries (like the -p set from the cmdline)

Reply:
Added map of phonebooks to MultiPhonebook.  Replace the map entries when updating DNS Bootstrap entries for SRV records.